traffic_stops/static/js/app/states/md/ContrabandHitRate.js

```diff
@@ -0,0 +1,188 @@
+          total[key] += year[key];
+        });
+      });
+
+      // push to end of array
+      total["year"] = "Total";
+      arr.push(total);
+    };
+
+    if (raw.searches.length>0) getTotals(raw.searches);
+    if (raw.contraband.length>0) getTotals(raw.contraband);
+
+    // set cleaned-data to handler
+    this.set("data", {
+      years: years,
+      raw: raw
+    });
+  }
+});
+
+// dashboard visuals
+
+var ContrabandHitRateBar = VisualBase.extend({
+  defaults: {
+    showEthnicity: true,
+    width: 750,
+    height: 375
+  },
+
+  setDefaultChart: function () {
+    this.chart = nv.models.multiBarHorizontalChart()
+      .x((d) => d.label)
+      .y((d) => d.value)
+      .barColor((d, i) => Stops.colors[1])
+      .width(this.get("width"))
+      .height(this.get("height"))
+      .margin({top: 20, right: 50, bottom: 20, left: 180})
+      .forceY([0, 1])
+      .showLegend(false)
+      .showValues(true)
+      .tooltips(true)
+      .transitionDuration(350)
+      .showControls(false);
+
+    this.chart.yAxis
+        .axisLabel('Searches resulting in contraband-found')
+        .tickFormat(d3.format('%'));
+
+    this.chart.valueFormat(d3.format('%'));
+  },
+
+  drawStartup: function () {
+    // get year options for pulldown menu
+    var selector = $('<select>'),
+        year_options = this.data.years,
+        opts = year_options.map((v) => `<option value="${v}">${v}</option>`),
+        getData = () => {
+          var year = selector.val();
+          year = parseInt(year, 10) || year;
+          this.dataset =  this._getDataset(year);
+          this.drawChart();
+        };
+
+    selector
+      .append(opts)
+      .val("Total")
+      .on('change', getData)
+      .trigger('change');
+
+    $('<div>')
+      .html(selector)
+      .appendTo(this.div);
+
+    this.selector = selector;
+  },
+
+  drawChart: function () {
+
+    d3.select(this.svg[0])
+            .datum(this.dataset)
+            .attr('width', "100%")
+            .attr('height', "100%")
+            .attr('preserveAspectRatio', "xMinYMin")
+            .attr('viewBox', `0 0 ${this.get('width')} ${this.get('height')}`)
+            .call(this.chart);
+
+    nv.utils.windowResize(this.chart.update);
+  },
+
+  _getDataset: function (year) {
+    var raw = this.data.raw,
+        searches_arr = raw.searches.filter((v) => v.year===year),
+        contraband_arr = raw.contraband.filter((v) => v.year===year),
+        dataset = {
+            color: Stops.single_color,
+            key: "Contraband hit-rates",
+            values: []
+        },
+        items = Stops.ethnicities,
+        ratio;
+
+    if (searches_arr.length===1 && contraband_arr.length===1){
+
+      searches_arr = searches_arr[0];
+      contraband_arr = contraband_arr[0];
+
+      // build a bar for each ethnicity
+      items.forEach((ethnicity, i) => {
+        ratio = (contraband_arr[ethnicity] / searches_arr[ethnicity]) || 0;
+        if (!isFinite(ratio)) ratio = undefined;
```

Comment:
When searches = 0, contraband = 0, too (for example, Asian for agency 62), and that shows up on the graph as 100%. It should probably show up as either 0% or text that says "no searches."